test/integration/filter_manager_integration_test.cc

```diff
@@ -0,0 +1,329 @@
+#include "test/integration/filter_manager_integration_test.h"
+
+#include <regex>
+
+#include "test/integration/http_integration.h"
+#include "test/integration/integration.h"
+#include "test/integration/utility.h"
+#include "test/server/utility.h"
+#include "test/test_common/registry.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+// Auxiliary network filter that makes use of ReadFilterCallbacks::injectReadDataToFilterChain()
+// and WriteFilterCallbacks::injectWriteDataToFilterChain() methods outside of the context of
+// ReadFilter::onData() and WriteFilter::onWrite(), i.e. on timer event
+const std::string inject_data_outside_callback_filter = "inject-data-outside-filter-callback";
```

Comment:
no global non-POD objects, put them in test cases or a singleton. this is easily leading to static-initialization-fiasco.

Reply:
Fixed